registration-processor/post-processor/registration-processor-printing-stage/src/main/java/io/mosip/registration/processor/print/util/CredentialPartnerUtil.java

```diff
@@ -1,75 +1,147 @@
-    private static final String IDENTITY_ATTRIBUTE = "mosip.identification.attribute";
-    private static final String NO_MATCH_ISSUER = "mosip.registration.processor.print.issuer.noMatch";
+    private static final String VALUE_LABEL = "value";
+    private static final String LANGUAGE = "language";
 
     @Autowired
     private Environment env;
 
     @Autowired
     private Utilities utilities;
 
+    @Value("#{T(java.util.Arrays).asList('${mosip.mandatory-languages:}')}")
+    private List<String> mandatoryLanguages;
+
+    @Value("${mosip.identification.attribute}")
+    private String identityAttribute;
+
+    @Value("${mosip.registration.processor.print.issuer.noMatch}")
+    private String noMatchIssuer;
+
     @Value("#{${mosip.registration.processor.print.issuer.config-map:{}}}")
     private Map<String, String> credentialPartnerExpression;
 
-    public Set<String> getCredentialPartners(String regId, JSONObject identity) {
+    /**
+     * This map will hold the actual field names after resolving, using mapping JSON as keys and
+     * configured field names as values
+     */
+    private Map<String, String> requiredIDObjectFieldNamesMap;
+    /**
+     * Configured Id object fields
+     */
+    private List<String> requiredIdObjectFieldNames;
+
+    public List<String> getCredentialPartners(String regId, String registrationType, JSONObject identity) throws PacketManagerException, JSONException, ApisResourceAccessException, IOException, JsonProcessingException {
 
         regProcLogger.info(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
                 regId, "CredentialPartnerUtil::getCredentialPartners()::entry");
 
-        Set<String> filteredPartners = new HashSet<>();
-        if (credentialPartnerExpression.isEmpty()) {
-            regProcLogger.error(LoggerFileConstant.SESSIONID.toString(),
+        List<String> filteredPartners = new ArrayList<>();
+        if (credentialPartnerExpression == null || credentialPartnerExpression.isEmpty()) {
+            regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(),
                     LoggerFileConstant.REGISTRATIONID.toString(), regId,
                     PlatformErrorMessages.RPR_PRT_ISSUER_NOT_FOUND_IN_PROPERTY.name());
             return filteredPartners;
         }
-        regProcLogger.info(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
-                regId, "CredentialPartnerUtil::getCredentialPartners()::" + credentialPartnerExpression.toString());
 
-        String identityValue = utilities.getIdJsonByAttribute(identity, env.getProperty(IDENTITY_ATTRIBUTE));
+        Map<String, String> identityFieldValueMap = utilities.getPacketManagerService().getFields(regId,
+                requiredIDObjectFieldNamesMap.values().stream().collect(Collectors.toList()), registrationType, ProviderStageName.PRINTING);
```

Comment:
Should we not take the keys here ? Since value would be the same identity attribute ? We need the actual field names right ?